modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/utils/CredentialApiUtil.java

```diff
@@ -0,0 +1,319 @@
+        }
+        File yamlFile = new File(openApiDir, yamlFileName);
+        File htmlFile = new File(openApiDir, "credentialApi" + ".html");
+        try {
+            if (yamlFile.exists()) {
+                log.info("credentialApi is generating");
+                Thread.sleep(2000);
+                if (htmlFile.exists()) {
+                    return getCredentialApiUrl();
+                }
+            } else {
+                String apiDocsJson = getApiDocsJson(credentialMap);
+
+                try (FileWriter apiWriter = new FileWriter(yamlFile.getPath())) {
+                    apiWriter.write(apiDocsJson);
+                }
+                log.info("credentialApi yamlFile:{} create success.", yamlFile.getPath());
+            }
+
+            if (yamlFile.exists() && openApiUtil.downloadClientJar(openApiDir)) {
+                File jarPath = new File(openApiDir, "openapi-generator-cli.jar");
+                String comm = String.format("java -jar %s generate -g html2 "
+                        + "-i %s -o %s", jarPath.getPath(), yamlFile.getPath(), openApiDir);
+                Process exec = Runtime.getRuntime().exec(comm);
+                StringBuilder stdErrOut = new StringBuilder();
+                BufferedReader outputReader =
+                        new BufferedReader(new InputStreamReader(exec.getErrorStream()));
+                String line;
+                while ((line = outputReader.readLine()) != null) {
+                    stdErrOut.append(line);
+                    log.error(line);
+                }
+                exec.waitFor();
+                if (exec.exitValue() != 0) {
+                    log.error("Create credentialApi html file failed." + stdErrOut);
+                }
+                // Modify the file name to credentialApi.html
+                File tempHtmlFile = new File(openApiDir, "index.html");
+                if (tempHtmlFile.exists()) {
+                    if (tempHtmlFile.renameTo(htmlFile)) {
+                        log.info("Create credentialApi html file:{} success.",
+                                htmlFile.getName());
+                        if (htmlFile.exists()) {
+                            return getCredentialApiUrl();
+                        }
+                    }
+                }
+            }
+            return StringUtils.EMPTY;
+        } catch (IOException | InterruptedException ex) {
+            log.error("Create credentialApi html file error:", ex);
+            throw new RuntimeException("Create service credentialApi html file error.", ex);
+        } finally {
+            // Delete the temp file named credentialApi.yaml
+            if (yamlFile.exists()) {
+                log.info("Delete credentialApi temp yaml file success:{}", yamlFile
+                        .delete());
+            }
+        }
+    }
+
+    /**
+     * generate CredentialApiUtil for plugins.
+     */
+    @Async("taskExecutor")
+    public void generateCredentialApi(
```

Comment:
Add param Csp to generate his own API doc  with name  ${cspName}_credentialApi.html